src/main/java/bio/terra/service/dataset/flight/datadelete/CreateExternalTablesStep.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.common.exception.PdaoInvalidUriException;
+import bio.terra.common.exception.PdaoSourceFileNotFoundException;
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.model.DataDeletionTableModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.exception.InvalidFileRefException;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+
+public class CreateExternalTablesStep implements Step {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final DatasetService datasetService;
+    private final Storage storage;
+
+    private static Logger logger = LoggerFactory.getLogger(CreateExternalTablesStep.class);
+
+    public CreateExternalTablesStep(BigQueryPdao bigQueryPdao, DatasetService datasetService) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+        this.storage = StorageOptions.getDefaultInstance().getService();
+    }
+
+    private boolean fileExistsForTable(DataDeletionTableModel table) {
+        // TODO: support other file types (assuming header-less csv here)
+        String path  = table.getGcsFileSpec().getPath();
+        try {
+            return GcsPdao.getBlobFromGsPath(storage, path).exists();
```

Comment:
This won't work if there's a wildcard in the path. When I added wildcard support on the ingest side, we [stripped away](https://github.com/DataBiosphere/jade-data-repo/pull/315/files#diff-536f43809cfb93169e496e7813e081daL60-R55) most up-front validation, and [caught the "not found" case](https://github.com/DataBiosphere/jade-data-repo/pull/315/files#diff-bdcc1fe50c17d33d41a93d142e860a12R355-R357) after it was raised by BigQuery.